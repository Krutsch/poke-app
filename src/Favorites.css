.favorite {
  display: grid;
  margin-top: 1em;
  grid-template-columns: repeat(auto-fit, minmax(215px, max-content));
  justify-content: center;
  grid-gap: 2em;
}
