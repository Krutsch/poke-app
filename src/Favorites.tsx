import React from "react";
import Pokemon from "./Pokemon";
import { useState, useEffect } from "react";
import "./Favorites.css";

function Favorites() {
  const [favorites, setFavorites] = useState<Array<string>>([]);

  useEffect(() => {
    const favStorage = window.localStorage.getItem("favorites") || "";
    const favs = Array.from(favStorage.split(" ")).filter(fav => !!fav);
    setFavorites(favs);
  }, []);
  return (
    <div className="favorite">
      {!favorites.length && "You selected no Favorites yet!"}

      {favorites.map((name, index) => (
        <Pokemon key={index} identifier={name}></Pokemon>
      ))}
    </div>
  );
}

export default Favorites;
