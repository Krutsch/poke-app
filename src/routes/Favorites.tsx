import React, { useState, useEffect } from "react";
import "../styles/Favorites.css";
import Pokemon from "../components/Pokemon";

function Favorites() {
  const [favorites, setFavorites] = useState<Array<string> | null>(null);

  useEffect(() => {
    const favStorage = window.localStorage.getItem("favorites") || "";
    const favs = Array.from(favStorage.split(" ")).filter(fav => !!fav);
    setFavorites(favs);
  }, []);

  return (
    <div className="favorite">
      {favorites !== null && favorites.length
        ? favorites.map(name => (
            <Pokemon key={name} identifier={name}></Pokemon>
          ))
        : "You selected no Favorites yet!"}
    </div>
  );
}
Favorites.whyDidYouRender = true;
export default Favorites;
