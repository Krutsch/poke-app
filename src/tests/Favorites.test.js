import React from "react";
import Favorites from "../routes/Favorites";
import ShallowRenderer from "react-test-renderer/shallow";

it("renders correctly", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Favorites />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
