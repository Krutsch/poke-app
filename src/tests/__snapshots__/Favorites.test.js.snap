// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="favorite"
>
  You selected no Favorites yet!
</div>
`;
